experimental/yarpl/include/yarpl/observable/Observers.h

```diff
@@ -12,15 +12,28 @@ namespace observable {
 
 /// Helper methods for constructing subscriber instances from functions:
 /// one, two, or three functions (callables; can be lamda, for instance)
-/// may be specified, corresponding to onNext, onError and onSubscribe
+/// may be specified, corresponding to onNext, onError and onComplete
 /// method bodies in the subscriber.
 class Observers {
+ private:
+  /// Defined if Next, Error and Complete are signature-compatible with
+  /// onNext, onError and onComplete subscriber methods respectively.
+  template <
+      typename T,
+      typename Next,
+      typename Error = void (*)(std::exception_ptr),
+      typename Complete = void (*)()>
+  using EnableIfCompatible =
+    typename std::enable_if<
+            std::is_callable<Next(T), void>::value &&
+            std::is_callable<Error(const std::exception_ptr), void>::value &&
```

Comment:
I know that this is copied from elsewhere, but I don't think that `const` in `const std::exception_ptr` is necessary?